csharp/AppEncryption/AppEncryption.Tests/AppEncryption/Kms/AwsKeyManagementServiceImplTest.cs

```diff
@@ -363,8 +364,6 @@ private void TestEncryptKeySuccessful()
             {
                 { EncryptedKey, Convert.ToBase64String(encryptedKey) },
                 {
-                    // For some reason we have to use ConcurrentBag here. Likely due to internal ConcurrentBag list
-                    // structure failing to compare against regular List?
                     KmsKeksKey, new ConcurrentBag<object>
```

Comment:
Can use a `List<object>` here and drop the `.ToList()` on L376

Reply:
Potentially, yes. The reason for doing it this was to keep the process of the envelope creation the same as the KMS implementation. https://github.com/godaddy/asherah/blob/e76a2c0cdf1cd9edee5ff64132130a78d127cbec/csharp/AppEncryption/AppEncryption/Kms/AwsKeyManagementServiceImpl.cs#L149

Also, I think  that using a List directly caused the test to fail intermittently in the past, as mentioned in the now deleted comment. Since the test now does exactly what we do in the code, the comment felt unnecessary.

Reply:
Discussed offline. Using a List<object> for the test